File: Python/Intersection_of_Linked_Lists.py
def LinkedListIntersection(headA, headB):
        #Save heads of a and b
        a, b = headA, headB
        #As long as both aren't equal (they'll be equal if the have an intersection
        # of if they both reach the end of the linked list)
        while a != b:
            #Swap the values if they've become null, else move onto next value
            a = headB if not a else a.next
            b = headA if not b else b.next
        return a

File: Java/Intersection_of_Linked_Lists.cpp
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode a = headA;
        ListNode b = headB;
        while (a != b){
            if(a == null) a = headB;
            else a = a.next;
            if(b == null) b = headA;
            else b = b.next;
        };
        return a;
        
    }
}